libconcurrent/includes/pool.h

```diff
@@ -3,22 +3,42 @@
 
 #include <primitives.h>
 
-typedef struct HalfPoolStruct {
-    char *heap;
-    int index;
-    int obj_size;
-} HalfPoolStruct;
+typedef struct BlockObject {
+    struct BlockObject *next;
+} BlockObject;
+
+typedef struct PoolBlockMetadata {
+    uint32_t object_size;
+    uint32_t entries;
+    uint32_t free_entries;
+    uint32_t cur_entry;
+    struct PoolBlock *next;
+    struct PoolBlock *back;
+} PoolBlockMetadata;
+
+typedef struct PoolBlock {
+    PoolBlockMetadata metadata;
+    char heap[];
+} PoolBlock;
 
 typedef struct PoolStruct {
-    char *heap;
-    int index;
-    int obj_size;
-    char pad[PAD_CACHE(sizeof(HalfPoolStruct))];
+    uint32_t obj_size;
```

Comment:
some implementations use `size_t` for variables that represent sizes

Reply:
This pool implementation is not able to support efficiently objects of more than 32 bits.
The size_t in some architectures may be of 64 bits ( ? ) and may let the user to ask for objects of size more than 32 bits, which is not a good idea in my opinion.